plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/TrinoFileStatusRemoteIterator.java

```diff
@@ -13,33 +13,62 @@
  */
 package io.trino.plugin.hive.fs;
 
+import io.trino.filesystem.FileEntry;
 import io.trino.filesystem.FileIterator;
+import jakarta.annotation.Nullable;
 
 import java.io.IOException;
+import java.util.NoSuchElementException;
+import java.util.function.Predicate;
 
 import static java.util.Objects.requireNonNull;
 
 public class TrinoFileStatusRemoteIterator
         implements RemoteIterator<TrinoFileStatus>
 {
     private final FileIterator iterator;
+    private final Predicate<FileEntry> filterPredicate;
+    @Nullable
+    private TrinoFileStatus nextElement;
 
-    public TrinoFileStatusRemoteIterator(FileIterator iterator)
+    public TrinoFileStatusRemoteIterator(FileIterator iterator, Predicate<FileEntry> filterPredicate)
+            throws IOException
     {
         this.iterator = requireNonNull(iterator, "iterator is null");
+        this.filterPredicate = filterPredicate;
```

Comment:
`this.filterPredicate = requireNonNull(filterPredicate, "filterPredicate is null");`

Reply:
same here; will fix